### ISSUE TITLE
feature request: workspace dependencies

### ISSUE THREAD

**Issue by tamird** [mitsuhiko/rye#734]
In a monorepo where a dependency is shared across multiple packages, it would be nice to maintain the version in one place. This is akin to cargo [workspace depdencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace).

**Comment by mitsuhiko** [comment]
I agree that this would be nice. That said, it would quite likely imply that `pyproject.toml` rewrites would be needed in some situations. `Cargo.toml` gets away with this because a) on publish rewrite happens and b) workspace dependencies are generally supported. Here the issue is that rye better not diverge too much from what other tools expect. For instance I do not feel comfortable putting proprietary extensions on top of a `project.dependencies` entry.

So the desire is real and I am in favor of this, but I am not yet sure how it can be done.

**Comment by minato-ellie** [comment]
PDM seems to be able to achieve this.[Use PDM to manage a monorepo](https://pdm-project.org/latest/usage/advanced/#use-pdm-to-manage-a-monorepo)

For my use case, I would like to at least have a mark in pyproject.toml. If it exists in the workspace, skip installation; otherwise, install it from PyPI. So I can use a mono repo for development and publish them separately to PyPI.
